Assets/MRTK/Providers/WindowsMixedReality/XR2018/Controllers/WindowsMixedRealityArticulatedHand.cs

```diff
@@ -90,88 +90,90 @@ private SpatialInteractionManager SpatialInteractionManager
 
         #region Update data functions
 
+        private static readonly ProfilerMarker UpdateControllerPerfMarker = new ProfilerMarker("[MRTK] WindowsMixedRealityArticulatedHand.UpdateController");
+
         /// <inheritdoc />
         public override void UpdateController(InteractionSourceState interactionSourceState)
         {
-            if (!Enabled) { return; }
-
-            Profiler.BeginSample("[MRTK] WindowsMixedRealityArticulatedHand.UpdateController");
+            using (UpdateControllerPerfMarker.Auto())
+            {
+                if (!Enabled) { return; }
 
-            base.UpdateController(interactionSourceState);
+                base.UpdateController(interactionSourceState);
 
-            UpdateHandData(interactionSourceState);
+                UpdateHandData(interactionSourceState);
 
-            for (int i = 0; i < Interactions?.Length; i++)
-            {
-                switch (Interactions[i].InputType)
+                for (int i = 0; i < Interactions?.Length; i++)
                 {
-                    case DeviceInputType.IndexFinger:
-                        handDefinition?.UpdateCurrentIndexPose(Interactions[i]);
-                        break;
+                    switch (Interactions[i].InputType)
+                    {
+                        case DeviceInputType.IndexFinger:
+                            handDefinition?.UpdateCurrentIndexPose(Interactions[i]);
+                            break;
+                    }
                 }
             }
-
-            Profiler.EndSample(); // UpdateController
         }
 
+        private static readonly ProfilerMarker UpdateHandDataPerfMarker = new ProfilerMarker("[MRTK] WindowsMixedRealityArticulatedHand.UpdateHandData");
```

Comment:
I _think_ this is only used in this context? To prevent a potential "field not used" warning.

```suggestion
#if WINDOWS_UWP || DOTNETWINRT_PRESENT
        private static readonly ProfilerMarker UpdateHandDataPerfMarker = new ProfilerMarker("[MRTK] WindowsMixedRealityArticulatedHand.UpdateHandData");
#endif // WINDOWS_UWP || DOTNETWINRT_PRESENT
```